django/db/backends/oracle/operations.py

```diff
@@ -29,12 +29,22 @@ class DatabaseOperations(BaseDatabaseOperations):
 DECLARE
     table_value integer;
     seq_value integer;
+    seq_name user_tab_identity_cols.sequence_name%%TYPE;
 BEGIN
+    BEGIN
+        SELECT sequence_name INTO seq_name FROM user_tab_identity_cols
+        WHERE  table_name = '%(table_name)s' AND
+               column_name = '%(column_name)s';
+        EXCEPTION WHEN NO_DATA_FOUND THEN
+            seq_name := '%(no_autofield_sequence_name)s';
+    END;
+
     SELECT NVL(MAX(%(column)s), 0) INTO table_value FROM %(table)s;
     SELECT NVL(last_number - cache_size, 0) INTO seq_value FROM user_sequences
-           WHERE sequence_name = '%(sequence)s';
+           WHERE sequence_name = seq_name;
     WHILE table_value > seq_value LOOP
-        seq_value := "%(sequence)s".nextval;
+        EXECUTE IMMEDIATE 'SELECT "'||seq_name||'".nextval FROM DUAL'
+        INTO seq_value;
```

Comment:
I know how it looks like, but there is no better way (in my opinion) to do this. We're establishing `seq_name` inside the trigger, therefore we cannot use bind variable for object name and we need to use string concatenation (see [doc](http://docs.oracle.com/database/121/LNPLS/dynamic.htm#LNPLS645)). I think that we can use
```SQL
seq_name := DBMS_ASSERT.ENQUOTE_NAME(seq_name, FALSE);
EXECUTE IMMEDIATE 'SELECT "'||seq_name||'".nextval FROM DUAL INTO seq_value;'
```
instead, but it seems to be overstatement for me.